src/b-splines/cubic.jl

```diff
@@ -210,38 +271,12 @@ This is the same system as `Quadratic{Free}` so we reuse the implementation
 """
 function prefiltering_system{T,TC,GT<:GridType}(::Type{T}, ::Type{TC}, n::Int,
                                                 ::Type{Cubic{Free}}, ::Type{GT})
-    prefiltering_system(T, TC, n, Quadratic{Free}, GT)
-end
-
-@inline cub(x) = x*x*x
-
-"""
-Build `rowspec`, `valspec`, `colspec` such that the product
-
-`out = rowspec * valspec * colspec` will be equivalent to:
-
-```julia
-out = zeros(n, n)
-
-for (i, j, v) in args
-    out[i, j] = v
-end
-```
-
-"""
-function _build_woodbury_specs{T}(::Type{T}, n::Int, args::Tuple{Int, Int, Any}...)
-    m = length(args)
-    rowspec = spzeros(T, n, m)
-    colspec = spzeros(T, m, n)
-    valspec = zeros(T, m, m)
```

Comment:
So this is where the problem is.

In `_build_woodbury_specs` we make `valspec` dense, but `WoodburyMatrices.sparse_factors` makes it sparse. 

This works without a problem:

```julia
julia> using WoodburyMatrices

julia> lupart = lufact!(Tridiagonal(ones(3), ones(4), ones(3)), Val{false});

julia> specs = WoodburyMatrices.sparse_factors(Float64, 4, (1, 3, 1.0), (4, 2, 1.0));

julia> Woodbury(lupart, specs...)
ERROR: The inverse of a sparse matrix can often be dense and can cause the computer to run out of memory. If you are sure you have enough memory, please convert your matrix to a dense matrix.
 in call at /Users/sglyon/.julia/v0.4/WoodburyMatrices/src/WoodburyMatrices.jl:41

julia> Woodbury(lupart, specs[1], full(specs[2]), specs[3])
4x4 WoodburyMatrices.Woodbury{Float64,Base.LinAlg.LU{Float64,Tridiagonal{Float64}},SparseMatrixCSC{Float64,Int64},SparseMatrixCSC{Float64,Int64},Array{Float64,2},Array{Float64,2}}:
 #undef  #undef  #undef  #undef
 #undef  #undef  #undef  #undef
 #undef  #undef  #undef  #undef
 #undef  #undef  #undef  #undef
```

@timholy what do you suggest we do here? Should we adjust `WoodburyMatrices.sparse_factors` to return a dense `valspec`?